controllers/core/openstackcontrolplane_controller.go

```diff
@@ -177,7 +186,50 @@ func (r *OpenStackControlPlaneReconciler) Reconcile(ctx context.Context, req ctr
 	return r.reconcileNormal(ctx, instance, helper)
 }
 
-func (r *OpenStackControlPlaneReconciler) reconcileNormal(ctx context.Context, instance *corev1beta1.OpenStackControlPlane, helper *helper.Helper) (ctrl.Result, error) {
+func (r *OpenStackControlPlaneReconciler) ensureDeploymentConfigMap(
+	ctx context.Context,
+	instance *corev1beta1.OpenStackControlPlane,
+	helper *helper.Helper,
+) error {
+	cmLabels := labels.GetLabels(instance, labels.GetGroupLabel("openstack"), map[string]string{})
+
+	customData := map[string]string{}
+
+	enabledServices := []string{}
+	// TODO: Add all supported services.
+	if instance.Spec.Nova.Enabled {
+		enabledServices = append(enabledServices, "nova")
+	}
+	if instance.Spec.Cinder.Enabled {
+		enabledServices = append(enabledServices, "cinder")
+	}
```

Comment:
We might want to add a helper function like https://github.com/openstack-k8s-operators/lib-common/pull/169